files/build_templates/qos_config.j2

```diff
@@ -240,13 +240,25 @@
 {% if 'type' in DEVICE_METADATA['localhost'] and DEVICE_METADATA['localhost']['type'] in backend_device_types and 'storage_device' in DEVICE_METADATA['localhost'] and DEVICE_METADATA['localhost']['storage_device'] == 'true' %}
             "dot1p_to_tc_map" : "AZURE",
 {% else %}
-{% if different_dscp_to_tc_map and port not in port_names_list_extra_queues and tunnel_qos_remap_enable %}
+{# Apply separated DSCP_TO_TC_MAP to uplink ports on ToR and Leaf #}
+{% if different_dscp_to_tc_map and tunnel_qos_remap_enable %}
+{% if ('type' in DEVICE_METADATA['localhost']) and (DEVICE_METADATA['localhost']['type'] == 'LeafRouter') and (port not in port_names_list_extra_queues) %}
             "dscp_to_tc_map"  : "AZURE_UPLINK",
+{% elif ('type' in DEVICE_METADATA['localhost']) and (DEVICE_METADATA['localhost']['type'] == 'ToRRouter') and (port in port_names_list_extra_queues) %}
```

Comment:
though 'tunnel_qos_remap_enable' will only be set in dualTor, as an extra precaution, instead of checking for type, check for 'subtype' == 'DualToR'

Reply:
FIxed. Thanks